common/grammar-parser.cpp

```diff
@@ -9,7 +9,7 @@
 namespace grammar_parser {
     // NOTE: assumes valid utf8 (but checks for overrun)
     // copied from llama.cpp
-    std::pair<uint32_t, const char *> decode_utf8(const char * src) {
+    static auto decode_utf8(const char * src) -> std::pair<uint32_t, const char *> {
```

Comment:
Let's avoid the `auto` in function declarations. The preferred way to simplify this code is to introduce a `struct pair_something` and use it instead of `std::pair`.

Reply:
It's just a trailing return type, not return type deduction. I would rather not make any changes to the types in this PR. Should I remove the trailing return type and format it differently?

Reply:
Yes, would like to keep the return type always in the front. We can do the type changes in separate PR